### PR TITLE
update videoJS to 5.4

### PR DESCRIPTION
**Pull request by jamesgorrie**
Updating video JS so we can start looking at updating the contribs (mostyle for ads as there are some performance stuff with Google IMA).

I would have like to have update to 5.5, but there is [a problem with the autoplay logic here](https://github.com/videojs/video.js/pull/2971#issuecomment-179201603) - I have raised [an issue](https://github.com/videojs/video.js/issues/3071) and will try to get that out as soon as.

Most contribs require 5.3 - so this is good for now.

### CODE REVIEW

File: static/src/javascripts/bower.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
       "lodash-amd": "bower_components/lodash-amd/compat/**",
       "domready": "bower_components/domready/ready.js",
       "when": "bower_components/when/es6-shim/Promise.js",
-      "video.js": "bower_components/video.js/dist/video.min.js",
+      "video.js": "bower_components/video.js/dist/video.js",
```
Comment (jamesgorrie) [inline static/src/javascripts/bower.json]: I'm not 100% sure if I have to update this elsewhere for `isProd`?

**Comment by jamesgorrie** [comment]
CC @Calanthe

File: common/app/model/Encoding.scala [hunk 2]
```diff
@@ -32,4 +32,4 @@ object EncodingOrdering extends Ordering[Encoding] {
 
   //Returns a negative integer, zero, or a positive integer as the first argument is less than, equal to, or greater than the second.
   def compare(x: Encoding, y: Encoding): Int = precedenceOf(x) - precedenceOf(y)
-}
```
Comment (rich-nguyen) [inline common/app/model/Encoding.scala]: ?

File: static/src/javascripts/bower.json (hunk 1 above)
Comment (rich-nguyen) [inline static/src/javascripts/bower.json]: nope, that is all

**Comment by rich-nguyen** [comment]
Hope that was easier than the last upgrade, which included a major version. if all the plugins behave fine, then :+1:

**Comment by rich-nguyen** [comment]
this reminds me, we need to talk about **tests**. Enough of this video wild west.

(we, as in the wider dotcom team)

**Comment by jamesgorrie** [comment]
A few of the plugins are actually broken at the moment (our own fullscreener for example).
Yip, all is well.

**Comment by jamesgorrie** [comment]
> Enough of this video wild west.

Damn it.

**Comment by jamesgorrie** [comment]
Saying that, I need to chat to @ScottPainterGNM about quality in general.

**Comment by ScottPainterGNM** [comment]
I approve of this conversation :smiling_imp:

**Comment by paperboyo** [comment]
Broken, broken they are!

File: static/src/javascripts/bower.json (hunk 1 above)
Comment (OliverJAsh) [inline static/src/javascripts/bower.json]: Out of curiosity, how come we changed this? Makes it easier to debug, that's for sure—and it's still minised later by Uglify through us :-)

File: common/app/model/Encoding.scala (hunk 2 above)
Comment (jamesgorrie) [inline common/app/model/Encoding.scala]: The IDE is doing it due to the editor config.

File: static/src/javascripts/bower.json (hunk 1 above)
Comment (jamesgorrie) [inline static/src/javascripts/bower.json]: [There was actually another place to change it](https://github.com/guardian/frontend/commit/4c13e708a35803dac3090876eb4ef7818ae4db0c)

Comment (rich-nguyen) [inline static/src/javascripts/bower.json]: apologies! And Olly makes a good point.

Comment (OliverJAsh) [inline static/src/javascripts/bower.json]: You will also have to update the require config: https://github.com/guardian/frontend/blob/9e110256e0a2f07b6f39a78de42fb4273f9eb362/grunt-configs/requirejs.js#L26

Comment (OliverJAsh) [inline static/src/javascripts/bower.json]: Must be horrible coming from a jspm world? :-P

**Pull request merged**